src/bun.js/api/bun/udp_socket.zig

```diff
@@ -0,0 +1,714 @@
+        }
+        const ret = this.doSend(globalThis, i);
+        if (ret) |val| {
+            // number of packets sent
+            return JSValue.jsNumber(val);
+        } else {
+            // exception
+            return .zero;
+        }
+    }
+
+    pub fn send(
+        this: *This,
+        globalThis: *JSGlobalObject,
+        callframe: *CallFrame,
+    ) callconv(.C) JSValue {
+        const arguments = callframe.arguments(3);
+        const dst: ?Destination = brk: {
+            if (this.connect_info != null) {
+                if (arguments.len == 1) {
+                    break :brk null;
+                }
+                if (arguments.len == 3) {
+                    globalThis.throwInvalidArguments("Cannot specify destination on connected socket", .{});
+                    return .zero;
+                }
+                globalThis.throwInvalidArguments("Expected 1 argument, got {}", .{arguments.len});
+                return .zero;
+            } else {
+                if (arguments.len != 3) {
+                    globalThis.throwInvalidArguments("Expected 3 arguments, got {}", .{arguments.len});
+                    return .zero;
+                }
+                break :brk .{
+                    .port = arguments.ptr[1],
+                    .address = arguments.ptr[2],
+                };
+            }
+        };
+
+        if (this.setupPacket(
+            globalThis,
+            0,
+            arguments.ptr[0],
+            dst,
+        )) |val| {
+            // we threw an exception
+            return val;
+        }
+
+        const sent = this.doSend(globalThis, 1);
+        if (sent) |val| {
+            // val can only be zero or one
+            return JSValue.jsBoolean(val > 0);
+        } else {
+            return .zero;
+        }
+    }
+
+    fn doSend(this: *This, globalThis: *JSGlobalObject, count: u16) ?c_int {
+        const res = this.socket.send(this.send_buf, count);
+        if (res == -1) {
+            const errno = @as(std.c.E, @enumFromInt(std.c._errno().*));
+            const err = bun.sys.Error.fromCode(errno, .sendmmsg);
+            globalThis.throwValue(err.toSystemError().toErrorInstance(globalThis));
+            return null;
+        }
+        return res;
+    }
+
+    const Destination = struct {
+        port: JSValue,
+        address: JSValue,
+    };
+
+    fn setupPacket(
+        this: *This,
+        globalThis: *JSGlobalObject,
+        index: usize,
+        data_val: JSValue,
+        dest: ?Destination,
+    ) ?JSValue {
+        if (index >= 1024) {
+            globalThis.throwInvalidArguments("Too many packets to send, maximum is 1024", .{});
+            return .zero;
+        }
+        const arg0 = data_val;
+        const payload = init: {
+            if (arg0.asArrayBuffer(globalThis)) |arrayBuffer| {
+                break :init arrayBuffer.slice();
+            } else if (bun.String.tryFromJS(arg0, globalThis)) |value| {
```

Comment:
This will call `.toString`, which means it will not throw if you pass it `{}` or `new Response` etc

We should probably instead check if `isString` and only then use tryFromJS

also, `tryFromJS` will only return null when it would otherwise have thrown. This would only happen when using something which throws in the `toString` function, like calling `Symbol("asd").toString()` - in which case, this would segfault because throwing an error inside an exception calls throw with a 0 JSValue